Apply default actions to default roles for new events/series

Arnei opened the pull request:
Fixes #1354.

An admin can configure actions that are automatically added to each role you add to an ACL in the ACL tab for series and events. However, they were not applied to default user role the ACL is initialized with. This adds the actions to default user role.

### How to test this

Configure additional actions in your Opencast configuration. Check `etc/listproviders/acl.additional.actions.properties` and `etc/listproviders/acl.default.create.properties` and configure an action there. Then try creating an event and a series.

![Bildschirmfoto vom 2025-07-02 16-26-43](https://github.com/user-attachments/assets/9f9c2e71-1984-443e-a85a-c1d6d608db52)

Arnei committed:
Apply default actions to default roles for new events/series

Fixes #1354.

An admin can configure actions that are automatically
added to each role you add to an ACL in the ACL tab
for series and events. However, they were
not applied to default user role the ACL
is initialized with. This adds the actions to
default user role.

Arnei committed:
Fix custom actions being lost on template switch for default user

When switching between ACL templates, we were keeping
the default user as intended, but the default user
lost their default actions (if configured). This fixes that.

snoesberger commented:
> > Additionally, when I change the template of an existing series, all the user ACLs are removed and only the editor's user ACL is added.
> 
> Afaik when switching templates, all changes made by the user are discarded. So after switching a template, only the roles specified by the template + the default user role are set. Wether that constitutes reasonable behaviour or not I cannot say. In any case, that discussion is probably better left for a different issue/PR.

In our case, we use the templates to regulate the access authorizations for Tobira. We use templates that contain roles which grant permissions based on users' organizational affiliations. For example, this means that we can only allow members of our institution to access certain series/events in Tobira.
Directly authorized users are owners of the corresponding series/events for us. Accordingly, we only want group authorizations to change when the template changes, not the owners. We have series with over 100 directly authorized users. If someone wants to change the template, for example, to make the series publicly available on Tobira instead of only to members of our institution, the 100+ users would lose their write access to the series.

By the way, in the old admin UI, user ACLs weren't removed when changing the template.

Arnei committed:
Fix  acl tab config settings not applying

The acl config defaults were not always fetched for the acl tab.
This should fix that.

snoesberger commented:
I found one other minor issue. The very first time I open the create dialogue, the default actions are not applied. As soon as I select a template, the default actions are applied, as they are when I open the create dialogue a second time.

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into acl-actions-missing-for-default-role

Arnei committed:
Fix default role actions not showing up for default user

For the acls, default actions can be configured to be added
automatically to each new role. However,
they were not added to the default role that gets automatically added.
This should fix that by removing some unnecessary react state.

marwyg commented:
So I tested the following things:
- deleting cookies and website data (in firefox when clicking on the lock symbol next to the url)
- opening series dialog the first time -> default actions appear ✅
- opening events dialog -> default actions appear ✅

- deleting cache again (step 1)
- opening events dialaog -> default actions don't appear ❌
- opening events dialog a second time -> default actions do appear

Arnei committed:
Fix default actions not being applied on first load

Not my favourite way to solve things, but with our current setup, this
is the way to go.

Ensures we have fetched the acl defaults from the backend
before opening the create event/series modal, so that Formik can be
properly initialized with said defaults
without the need for enabling reinitialization.

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into acl-actions-missing-for-default-role
